### PR TITLE
ROS2: Optical Flow, AR Tags, Handrail Detect

### PR DESCRIPTION
**Pull request by Pedro-Roque**
Adding handrail detect plugin.

This plugin had not been ported to Gazebo > 7. Probably it's not being used anymore? I ported the content to the current version of Gazebo and everything compiles fine.

### CODE REVIEW

File: simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc [hunk 1]
```diff
@@ -71,147 +80,122 @@ class GazeboSensorPluginHandrailDetect : public FreeFlyerSensorPlugin {
...
-        Eigen::Translation3d(
-          sensor_->Pose().Pos().X(),
-          sensor_->Pose().Pos().Y(),
-          sensor_->Pose().Pos().Z()) *
-        Eigen::Quaterniond(
-          sensor_->Pose().Rot().W(),
-          sensor_->Pose().Rot().X(),
-          sensor_->Pose().Rot().Y(),
-          sensor_->Pose().Rot().Z()));
+    Eigen::Affine3d wTb = (Eigen::Translation3d(GetModel()->WorldPose().Pos().X(), GetModel()->WorldPose().Pos().Y(),
```
Comment (marinagmoreira) [inline simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc:159]: why realign this if it was more readable before?

File: simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc [hunk 2]
```diff
@@ -71,147 +80,122 @@ class GazeboSensorPluginHandrailDetect : public FreeFlyerSensorPlugin {
...
+      (Eigen::Translation3d(sensor_->Pose().Pos().X(), sensor_->Pose().Pos().Y(), sensor_->Pose().Pos().Z()) *
+       Eigen::Quaterniond(sensor_->Pose().Rot().W(), sensor_->Pose().Rot().X(), sensor_->Pose().Rot().Y(),
+                          sensor_->Pose().Rot().Z()));
     Eigen::Affine3d wTc = wTb * bTc;
 
     // Initialize the camera paremeters
     static camera::CameraParameters cam_params(&config_, "perch_cam");
-    static camera::CameraModel camera(Eigen::Vector3d(0, 0, 0),
-      Eigen::Matrix3d::Identity(), cam_params);
+    static camera::CameraModel camera(Eigen::Vector3d(0, 0, 0), Eigen::Matrix3d::Identity(), cam_params);
```
Comment (marinagmoreira) [inline simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc:171]: no need to realign this

File: simulation/src/gazebo_sensor_plugin_optical_flow/gazebo_sensor_plugin_optical_flow.cc [hunk 3]
```diff
@@ -42,180 +49,137 @@
...
+      return;
     }
 
-    if (!config_.GetReal("rate", &rate_))
-      ROS_FATAL("Could not read the rate parameter.");
+    if (!config_.GetReal("rate", &rate_)) FF_FATAL("Could not read the rate parameter.");
 
-    if (!config_.GetReal("delay_camera", &delay_camera_))
-      ROS_FATAL("Could not read the delay_camera parameter.");
+    if (!config_.GetReal("delay_camera", &delay_camera_)) FF_FATAL("Could not read the delay_camera parameter.");
```
Comment (bcoltin) [inline simulation/src/gazebo_sensor_plugin_optical_flow/gazebo_sensor_plugin_optical_flow.cc:85]: Same realignment issues here

File: simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc [hunk 4]
```diff
@@ -71,147 +80,122 @@ class GazeboSensorPluginHandrailDetect : public FreeFlyerSensorPlugin {
...
     if (!config_.ReadFiles()) {
-        ROS_ERROR("Failed to read config files.");
-        return;
+      FF_ERROR("Failed to read config files.");
+      return;
     }
 
-    if (!config_.GetReal("rate", &rate_))
-      ROS_FATAL("Could not read the rate parameter.");
+    if (!config_.GetReal("rate", &rate_)) FF_FATAL("Could not read the rate parameter.");
```
Comment (bcoltin) [inline simulation/src/gazebo_sensor_plugin_handrail_detect/gazebo_sensor_plugin_handrail_detect.cc:87]: and here

**Comment by marinagmoreira** [comment]
we should open a separate issue as there is no handrail detect plugin starting in the description folder!

**Review by marinagmoreira — CHANGES_REQUESTED** [review]
Should minimize the style changes and then good to merge.

**Comment by Pedro-Roque** [comment]
... mostly due to auto-formatter being on, I'll fix them, sorry for the trouble!

**Comment by Pedro-Roque** [comment]
Done @marinagmoreira . I see there is a style check on commits, maybe we can add these styles to regarding line length.

**Review by marinagmoreira — APPROVED** [review]

**Pull request merged**